pages/allDonations.js

```diff
@@ -15,52 +15,93 @@ export default function AllDonations({
 
   return (
     <>
-      <h2> Spenden in deiner Nähe: </h2>
-      <input
-        type="text"
-        pattern="^[2-9][0-9][0-9][0-9][0-9]$"
-        title="Bitte 5-stellige PLZ eingeben"
-        value={zipCode}
-        onChange={(event) => setZipCode(event.target.value)}
-        placeholder="Nach deiner PLZ filtern..."
-      />
-      <StyledList>
-        {filteredDonations.map((donation) => {
-          return (
-            <DonationListItem
-              key={donation.id}
-              donation={donation}
-              donations={donations}
-              handleEditDonation={handleEditDonation}
-              myUserID={myUserID}
-            />
-          );
-        })}
-      </StyledList>
-
+      <StyledWrapper>
+        <StyledInput
+          type="text"
+          pattern="^[2-9][0-9][0-9][0-9][0-9]$"
```

Comment:
I think your regex pattern is really interesting, but why did you exclude postal codes that have 0 and 1 as first character? 👀  You could adjust this pattern and use this pattern instead: `^[0-9]+` :)